commons/src/test/java/org/sonarsource/analyzer/commons/EducationRuleLoaderTest.java

```diff
@@ -0,0 +1,275 @@
+    context = new Context();
+    newRepository = context.createRepository(RULE_REPOSITORY_KEY, "lang");
+    new RulesDefinitionAnnotationLoader().load(newRepository, new Class[]{TestRule.class});
+    newRule = newRepository.rule("MyRuleKey");
+  }
+
+  @Test
+  public void supports_education_rules_descriptions() {
+    assertThat(new EducationRuleLoader(SonarRuntimeImpl.forSonarQube(Version.create(9, 5), SonarQubeSide.SERVER, SonarEdition.DEVELOPER)).isEducationRuleDescriptionSupported()).isTrue();
+    assertThat(new EducationRuleLoader(SonarRuntimeImpl.forSonarQube(Version.create(9, 4), SonarQubeSide.SERVER, SonarEdition.DEVELOPER)).isEducationRuleDescriptionSupported()).isFalse();
+    assertThat(new EducationRuleLoader(SonarRuntimeImpl.forSonarQube(Version.create(9, 6), SonarQubeSide.SERVER, SonarEdition.SONARCLOUD)).isEducationRuleDescriptionSupported()).isTrue();
+    assertThat(new EducationRuleLoader(SonarRuntimeImpl.forSonarLint(Version.create(9, 6))).isEducationRuleDescriptionSupported()).isTrue();
+  }
+
+  @Test
+  public void supports_education_principles_metadata() {
+    assertThat(new EducationRuleLoader(SonarRuntimeImpl.forSonarQube(Version.create(9, 8), SonarQubeSide.SERVER, SonarEdition.DEVELOPER)).isEducationPrinciplesMetadataSupported()).isTrue();
+    assertThat(new EducationRuleLoader(SonarRuntimeImpl.forSonarQube(Version.create(9, 8), SonarQubeSide.SERVER, SonarEdition.SONARCLOUD)).isEducationPrinciplesMetadataSupported()).isTrue();
+    assertThat(new EducationRuleLoader(SonarRuntimeImpl.forSonarLint(Version.create(9, 8))).isEducationPrinciplesMetadataSupported()).isTrue();
+    assertThat(new EducationRuleLoader(SonarRuntimeImpl.forSonarQube(Version.create(9, 7), SonarQubeSide.SERVER, SonarEdition.DEVELOPER)).isEducationPrinciplesMetadataSupported()).isFalse();
+    assertThat(new EducationRuleLoader(SonarRuntimeImpl.forSonarLint(Version.create(9, 7))).isEducationPrinciplesMetadataSupported()).isFalse();
+  }
+
+  @Test
+  public void education_description_content_unsupported_product_runtime() throws IOException {
+    SonarRuntime invalidRuntime = SonarRuntimeImpl.forSonarQube(Version.create(9, 4), SonarQubeSide.SERVER, SonarEdition.DEVELOPER);
+    EducationRuleLoader educationRuleLoader = new EducationRuleLoader(invalidRuntime);
+    String testFileContent = getTestFileContent("valid/S100.html");
+    String fallbackDescription = educationRuleLoader.setEducationDescriptionFromHtml(newRule, testFileContent);
+    newRepository.done();
+    RulesDefinition.Rule rule = context.repository(RULE_REPOSITORY_KEY).rule("MyRuleKey");
+
+    assertThat(rule.ruleDescriptionSections()).isEmpty();
+    assertThat(fallbackDescription).isEqualTo(testFileContent);
+  }
+
+  @Test
+  public void education_description_content_unsupported_product_runtime_invalid_format() throws IOException {
+    SonarRuntime invalidRuntime = SonarRuntimeImpl.forSonarQube(Version.create(9, 4), SonarQubeSide.SERVER, SonarEdition.DEVELOPER);
```

Comment:
maybe use a supported version for this case, to make sure the behaviour is triggered by the invalidity?

Reply:
There's actually already another test that check that file with a supported version of the runtime.
This test was actually not useful, as it didn't test anything specific. I removed it :) :+1: